internal/storage/db/get.go

```diff
@@ -0,0 +1,261 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/kamencov/go-musthave-diploma-tpl/internal/customerrors"
+	"github.com/kamencov/go-musthave-diploma-tpl/internal/models"
+	"net/http"
+	"time"
+)
+
+func (d *DateBase) Get(query string, args ...interface{}) (*sql.Row, error) {
+	row := d.storage.QueryRow(query, args...)
+	if row == nil {
+		return nil, customerrors.ErrNotFound
+	}
+
+	return row, nil
+}
+
+func (d *DateBase) Gets(query string, args ...interface{}) (*sql.Rows, error) {
+	rows, err := d.storage.Query(query)
+	if err != nil {
+		return nil, err
+	}
+
+	return rows, err
+}
+
+func (d *DateBase) GetLoginID(login string) (int, error) {
+	var user int
+
+	query := `SELECT id FROM users WHERE login = $1`
+
+	row, err := d.Get(query, login)
+	if err != nil {
+		return -1, customerrors.ErrNotData
+	}
+
+	if err = row.Scan(&user); err != nil {
+		return -1, customerrors.ErrNotUser
+	}
+
+	return user, nil
+}
+
+func (d *DateBase) GetUserByAccessToken(order string, login string, now time.Time) error {
```

Comment:
На мой взгляд не совсем удачное название для метода. Во-первых get происходит не по access token, а по логину. Во-вторых поиск происходит не только пользователя. В третьих еще происходит с сохранение заказа. Мне кажется эту логику стоит перенести на уровень сервиса, тогда у него появится смысл, а слой работы с БД не будет перегружен бизнес логикой.

```suggestion
func (d *DateBase) GetUserByAccessToken(order string, login string, now time.Time) error {
```